fusesoc/main.py

```diff
@@ -298,6 +301,127 @@ def core_sign(fs, args):
     print(f"{sigfile} created")
 
 
+def guess_provider():
```

Comment:
Since there is quite a lot of new code, I would prefer to have this in a separate `publish.py`. `main.py` is already way too large for its own good.